### PR TITLE
Unify `border-radius` behavior

### PR DESCRIPTION
**Pull request by delvh**
## Changes
- no more hardcoded `border-radius`es (apart from `0`)
- no more value inconsistencies
- no more guessing what pixel value you should use
- two new variables:
  - `--border-radius-medium` (for elements where the normal border radius does not suffice)
  - `--border-radius-circle` (for displaying circles)

### CODE REVIEW

**Comment by delvh** [comment]
I've tested every occurrence where I could find/reproduce the element.
Some changes are untested (i.e. because I don't own an IOS device), but I doubt this PR introduces a regression.

File: web_src/css/base.css [hunk 1]
```diff
@@ -17,6 +17,8 @@
   --octicon-chevron-right: url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="16" height="16"><path d="M6.22 3.22a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L9.94 8 6.22 4.28a.75.75 0 0 1 0-1.06Z"></path></svg>');
   /* non-color variables */
   --border-radius: 0.28571429rem;
```
Comment (silverwind) [inline web_src/css/base.css]:
```suggestion
  --border-radius: 4px;
```

With 14px base font size, this comes out as `3.9998px`, so let's make it integer.

Comment (delvh) [reply web_src/css/base.css]: That's simply what we got from Fomantic, so I mean we CAN change it but we don't need to…

**Comment by silverwind** [comment]
Looks good overall. I hope 8px is not too much for the large ones that were previously 6px.

**Review by wxiaoguang — APPROVED** [review]
Agree to use `6px` instead of `calc(2 * var(--border-radius))`

File: web_src/css/base.css [hunk 2]
```diff
@@ -16,7 +16,9 @@
   --checkbox-mask-indeterminate: url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="16" height="16"><path fill-rule="evenodd" d="M2 7.75A.75.75 0 012.75 7h10a.75.75 0 010 1.5h-10A.75.75 0 012 7.75z"></path></svg>');
   --octicon-chevron-right: url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="16" height="16"><path d="M6.22 3.22a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L9.94 8 6.22 4.28a.75.75 0 0 1 0-1.06Z"></path></svg>');
   /* non-color variables */
-  --border-radius: 0.28571429rem;
+  --border-radius: 4px;
+  --border-radius-large: calc(2 * var(--border-radius));
+  --circle-border: 100%;
```
Comment (wxiaoguang) [inline web_src/css/base.css]:
```suggestion
  --border-radius-circle: 100%;
```

The name could be this? Considering other vars are called `--border-radius-xxx`

Comment (delvh) [reply web_src/css/base.css]: Done

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
I wonder how we will name next step after `large`: `huge`? 😆

**Comment by silverwind** [comment]
Maybe follow tailwind and call it medium?

```
rounded | border-radius: 0.25rem; /* 4px */
rounded-md | border-radius: 0.375rem; /* 6px */
rounded-lg | border-radius: 0.5rem; /* 8px */
```

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
Yes, now we are free to add `large` should the need arise 😆

**Comment by wxiaoguang** [comment]
Regression:

image


ps: I would say I still prefer keeping some special values in code, instead of forcing everything into a variable.

**Comment by delvh** [comment]
What was the behavior before?
I doubt it was much different:
The text should still have stretched the circle.
Everything else does not make much sense, unless you want to tell me that the text previously overflowed the circle.

**Comment by wxiaoguang** [comment]
> What was the behavior before?

Before:

image


After:

image

**Comment by silverwind** [comment]
The previous `17px` radius was very deliberatly chosen to exactly prevent this kind of stretching. In hindsight, I should not have approved this.

**Comment by wxiaoguang** [comment]
->  Fix notification circle (border-radius) #26794

**Pull request merged**